tests/lax_scipy_test.py

```diff
@@ -190,6 +190,19 @@ def testLogSumExpNans(self):
         result = lsp_special.logsumexp(1.0, b=1.0)
         self.assertEqual(result, 1.0)
 
+  @jtu.sample_product(
+    shape=[(0,), (1,), (2,), (3,), (4,), (5,)],
+    dtype=float_dtypes,
+  )
+  def testLogSumExpWhere(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    x = rng(shape, dtype)
+    rng = jtu.rand_bool(self.rng())
+    mask = rng(shape, bool)
+    y_expected = osp_special.logsumexp(x[mask]) if mask.any() else -jnp.inf
+    y_actual = lsp_special.logsumexp(x, where=mask)
+    self.assertAllClose(y_expected, y_actual)
```

Comment:
It looks like this breaks in X64 mode, because the dtypes of the outputs don't match when the input is float32. I'd suggest passing `check_dtypes=False` here:
```suggestion
    self.assertAllClose(y_expected, y_actual, check_dtypes=False)
```